src/core/stream/index.ts

```diff
@@ -219,22 +220,31 @@ export default function Stream({
     log.debug("initial time calculated:", initialTime);
 
     const reloadStreamSubject$ = new Subject<void>();
-    (window as any).reload = reloadStreamSubject$;
-    const reloadStream$ : Observable<IStreamEvent> = reloadStreamSubject$.pipe(
-      switchMap(() => {
+    const reloadStream$ : Observable<IStreamEvent> = reloadStreamSubject$
+      .pipe(switchMap(() => {
         const currentPosition = videoElement.currentTime;
         const isPaused = videoElement.paused;
         return openMediaSource(videoElement).pipe(
           mergeMap(newMS => loadStream(newMS, currentPosition, !isPaused)),
+          tap(({ type }) => {
+            if (type === "needs-stream-reload") {
+              reloadStreamSubject$.next();
+            }
+          }),
           startWith(EVENTS.reloadingStream())
         );
-      })
-    );
+      }));
 
     const initialLoad$ = observableConcat(
       observableOf(EVENTS.manifestReady(abrManager, manifest)),
-      loadStream(mediaSource, initialTime, autoPlay)
-        .pipe(takeUntil(reloadStreamSubject$))
+      loadStream(mediaSource, initialTime, autoPlay).pipe(
+        takeUntil(reloadStreamSubject$),
+        tap(({ type }) => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 229. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=286&issues=AWVIuc9v2xJdnw6QbVlk&open=AWVIuc9v2xJdnw6QbVlk)

[ISSUE_KEY]: AWVIuc9v2xJdnw6QbVlk